bin/metababel

```diff
@@ -23,6 +23,31 @@ def sanitize(str)
   str.gsub(/[^0-9A-Za-z-]/, '_')
 end
 
+def btx_match?(str, pattern_or_str)
+  # String beginning and ending in '/' are treated as regexp.
+  match = pattern_or_str.match(%r{^/(?<expr>.+)/$})
+  match ? str.match?(match['expr']) : str == pattern_or_str
+end
+
+def btx_event_match(evt_name, evt_args, match_name, match_args)
+  # Return nil if the event and args do not match.
+  # Return a list otherwise.
+
+  # If match_args.length > evt_args.length, a match_arg must match one evt_arg
+  # twice, but that is not allowed. 
+  if btx_match?(evt_name, match_name) and evt_args.length >= match_args.length
+    matched_args = match_args.filter_map do |match_arg|
+      matched_event_args = evt_args.find_all { |evt_arg| btx_match?(evt_arg.name, match_arg) }
+    end.flatten
+
+    unless matched_args.uniq.length == match_args.length
+      raise "Memeber regex matched more than one event member, got '#{matched_args.map(&:name).join(', ') }' from '#{match_args.join(', ') }'."
+    end
```

Comment:
You need to check also that `matched_args.uniq.length == matched_args.length`, so you are sure you don't match the same twice

Reply:
I think it is already there, isn't it?

Reply:
What do you mean?  You want 3 tests `matched_args.uniq.length == match_args.length == matched_args.length`